File: 2020-01/31/chris.java
public class _31_Best {
    public int maxProfit(final int[] prices) {
        int totalProfit = 0;

        final int loop = prices.length - 1;

        for (int i = 0; i < loop; i++) {
            final int todayPrice = prices[i];
            final int tomorrowPrice = prices[i + 1];

            if (todayPrice < tomorrowPrice) {
                totalProfit += tomorrowPrice - todayPrice;
            }
        }

        return totalProfit;
    }
}

File: 2020-01/31/chris.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        if prices.len() == 0 {
            return 0;
        }

        let mut totalProfit = 0;

        let looop = prices.len() - 1;
        for i in 0..looop {
            let todayPrice = unsafe { prices.get_unchecked(i) };
            let tomorrowPrice = unsafe { prices.get_unchecked(i + 1) };

            if todayPrice < tomorrowPrice {
                totalProfit += (tomorrowPrice - todayPrice);
            }
        }

        return totalProfit;
    }
}
